### PR TITLE
ames: forward iff we're a galaxy

### LINKED ISSUE #492: ames: forward packets iff we're a galaxy
See "change 1" here: https://github.com/urbit/urbit/pull/6700

### PR DESCRIPTION
**Pull request by matthew-levan**
Resolves #492.

### CODE REVIEW

**Comment by matthew-levan** [comment]
Where is the best place to add a helper function to determine which identity type (comet, moon, planet, star, galaxy) we are, given `c3_d *who_d`? We need such a function to set the new "forwarding enabled" flag `sat_u.for_o` in the `u3_ames` struct.

**Comment by philipcmonk** [comment]
So far we've just done `u3do("clan:title", u3k(ship))`.  Wouldn't be hard to add a clan function (especially if we need to call it before the loom is set up), not sure where it fits, maybe vortex.c?

**Comment by matthew-levan** [comment]
Thanks @philipcmonk. How's this look now?

File: pkg/vere/io/ames.c [hunk 1]
```diff
@@ -1993,7 +1994,9 @@ _ames_hear(u3_ames* sam_u,
      && (  (pac_u->pre_u.rec_d[0] != sam_u->pir_u->who_d[0])
         || (pac_u->pre_u.rec_d[1] != sam_u->pir_u->who_d[1]) ) )
   {
-    _ames_try_forward(pac_u);
+    if ( _(sam_u->sat_u.for_o) ) {
```
Comment (philipcmonk) [inline pkg/vere/io/ames.c:1997]: I think we mostly do `== c3y` instead of `_` because it's a bit clearer.

File: pkg/vere/io/ames.c [hunk 2]
```diff
@@ -2458,6 +2461,11 @@ u3_ames_io_init(u3_pier* pir_u)
   sam_u->fig_u.see_o = c3y;
   sam_u->fig_u.fit_o = c3n;
 
+  //  enable forwarding on galaxies only
+  u3_noun who = u3i_chubs(2, sam_u->pir_u->who_d);
+  u3_noun rac = u3do("clan:title", u3k(who));
```
Comment (philipcmonk) [inline pkg/vere/io/ames.c:2466]: `who` is leaked here.  Probably easiest to just remove the `u3k` to transfer it into `u3do`

**Review by philipcmonk — CHANGES_REQUESTED** [review]

**Review by philipcmonk — APPROVED** [review]
Looks good!

**Pull request merged**